cordapp-contracts-states/src/main/kotlin/com/negotiation/StatesAndContracts.kt

```diff
@@ -30,11 +30,37 @@ open class ProposalAndTradeContract : Contract {
                 "There are zero input states" using (tx.inputStates.isEmpty())
                 "There is no timestamp" using (tx.timeWindow == null)
             }
+
             is Commands.Accept -> requireThat {
-                // TODO: Add contractual logic.
+                val input = tx.inputsOfType<ProposalState>().single()
+                val output = tx.outputsOfType<TradeState>().single()
+
+                "There is exactly one output" using (tx.outputStates.size == 1)
+                "There is exactly one input" using (tx.inputStates.size == 1)
+
+                "The amount should be same after accepting the proposal" using (output.amount == input.amount)
+
+                "The buyer and seller are unmodified in the output" using (input.buyer == output.buyer && input.seller == output.seller)
+
+                "The proposer is a required signer" using (cmd.signers.contains(input.proposer.owningKey))
+                "The proposee is a required signer" using (cmd.signers.contains(input.proposee.owningKey))
             }
+
             is Commands.Modify -> requireThat {
-                // TODO: Add contractual logic.
+                val output = tx.outputsOfType<ProposalState>().single()
+                val input = tx.inputsOfType<ProposalState>().single()
+
+                "There is exactly one output" using (tx.outputStates.size == 1)
+                "There is exactly one input" using (tx.inputStates.size == 1)
+                "The new amount and old amount should not be same" using (output.amount != input.amount)
+
+                "The proposer is a required signer in the input" using (cmd.signers.contains(input.proposer.owningKey))
```

Comment:
I'd rather replace these four lines with 2 checks:
1. That the proposer and proposee are unchanged
2. That the proposer and proposee are required signers in the output